apps/explorer/lib/explorer/chain/transaction.ex

```diff
@@ -624,4 +663,23 @@ defmodule Explorer.Chain.Transaction do
       limit: 1
     )
   end
+
+  def get_token_name(%__MODULE__{gas_currency_hash: nil}), do: {:error, :not_found}
+  def get_token_name(%__MODULE__{
```

Comment:
Of course not an expert in Ecto, but is there no native way of expressing this relationship so that Ecto can possibly eager load it in a join?

Reply:
Hi @nambrot. I tried to implement this with [ecto's preload](apps/explorer/lib/explorer/chain/transaction.ex ), but I could not manage to get it working. I am not an elixir/ecto expert neither, so because this way works well recurring to an additional method I did not investigate much more the other solution.
I can try to modify the implementation to add the `preload` directive to the correct place, but maybe we can include that change in a new PR?